drivers/sdcard/sdcard.py

```diff
@@ -159,27 +164,19 @@ def cmd(self, cmd, arg, crc, final=0, release=True):
         self.spi.write(b'\xff')
         return -1
 
-    def cmd_nodata(self, cmd):
-        self.spi.write(cmd)
-        self.spi.read(1, 0xff) # ignore stuff byte
-        for _ in range(_CMD_TIMEOUT):
-            if self.spi.read(1, 0xff)[0] == 0xff:
-                self.cs(1)
-                self.spi.write(b'\xff')
-                return 0    # OK
-        self.cs(1)
-        self.spi.write(b'\xff')
-        return 1 # timeout
-
     def readinto(self, buf):
         self.cs(0)
 
         # read until start byte (0xff)
-        while self.spi.read(1, 0xff)[0] != 0xfe:
-            pass
+        while True:
+            self.spi.readinto(self.tokenbuf, 0xff)
+            if self.tokenbuf[0] == 0xfe:
+                break
 
         # read data
-        mv = self.dummybuf_memoryview[:len(buf)]
+        mv = self.dummybuf_memoryview
+        if len(buf) != len(mv):
```

Comment:
Only allocate the new `memoryview` when the size is not 512 bytes. It almost always is 512 bytes, except while reading the CSD block.

Reply:
Looks good.  Ultimately there'd be a way to adjust a memory view's offset/length without allocating memory, but that doesn't exist yet.

Reply:
Yes that would be very useful to avoid garbage.